### PR TITLE
fix: [CDS-59440]: fixed to get the details from template

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx [hunk 1]
```diff
@@ -233,18 +237,59 @@ function BarrierInputStep({ inputSetData, allowableTypes, stepViewType }: Barrie
...
+  }, [pipeline?.pipeline?.template?.templateRef])
+
+  React.useEffect(() => {
+    if (pipelineTemplateResponse?.data && !pipelineTemplateLoading) {
+      const templateSpec = parse<{ template: { spec: StepElementConfig } }>(
+        defaultTo(pipelineTemplateResponse?.data?.yaml, '')
+      )
+      setPipelineTemplate(templateSpec as any)
+    }
+  }, [pipelineTemplateResponse?.data, pipelineTemplateLoading])
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx:278]: Can we remove pipelineTemplate and useEffect and use memoized value 
```suggestion
const pipelineTemplate = React.useMemo(() => {
  if (pipelineTemplateResponse?.data && !pipelineTemplateLoading) {
    const templateSpec = parse<{ template: { spec: StepElementConfig } }>(
      defaultTo(pipelineTemplateResponse?.data?.yaml, '')
    )
    return templateSpec as any;
  }
  return undefined;
}, [pipelineTemplateResponse?.data, pipelineTemplateLoading]);
```

**Comment by sahithikolichala** [comment]
retrigger jest

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx [hunk 2]
```diff
@@ -233,18 +237,59 @@ function BarrierInputStep({ inputSetData, allowableTypes, stepViewType }: Barrie
       parentEntityRepoName: repoName
     }
   })
+
+  const {
+    data: pipelineTemplateResponse,
+    refetch: refetchPipelineTemplate,
+    loading: pipelineTemplateLoading
+  } = useGetTemplate({
+    templateIdentifier: pipeline?.pipeline?.template?.templateRef || '',
```
[the remark this answers is not in the document]
Comment (sahithikolichala) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx]: I have modified the parameters for the api to the following
https://github.com/harness/harness-core-ui/pull/15435/files#diff-ebe086aaebb3f3e05289fe901bcd99604d1c73d460145e99024184cc47dc918fR263

[the remark this answers is not in the document]
Comment (sahithikolichala) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx]: posted recordings of the testing here 
https://harness.slack.com/archives/C01K0MDQXL3/p1682536010935519

**Comment by sahithikolichala** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx [hunk 3]
```diff
@@ -218,9 +231,13 @@ function BarrierInputStep({ inputSetData, allowableTypes, stepViewType }: Barrie
     PipelineType<InputSetPathProps> & { accountId: string }
   >()
   const { repoIdentifier, branch, connectorRef, repoName } = useQueryParams<GitQueryParams>()
+  const projectQueryParams = useParams<ProjectPathProps>()
+
   const { expressions } = useVariablesExpression()
 
   const [pipeline, setPipeline] = React.useState<{ pipeline: PipelineInfoConfig } | undefined>()
+  const stepTemplateVersionLabel = defaultTo(pipeline?.pipeline?.template?.versionLabel, '')
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx:239]: This name is incorrect as this is not a step template. We can rename "stepTemplateVersionLabel"

**Review by samarth-sharma31 — DISMISSED** [review]

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx [hunk 4]
```diff
@@ -233,18 +250,64 @@ function BarrierInputStep({ inputSetData, allowableTypes, stepViewType }: Barrie
...
+    refetch: refetchPipelineTemplate,
+    loading: pipelineTemplateLoading
+  } = useGetTemplate({
+    templateIdentifier: getIdentifierFromValue(pipeline?.pipeline?.template?.templateRef || ''),
+
+    queryParams: {
+      ...getScopeBasedProjectPathParams(projectQueryParams, templateScope),
+      versionLabel: pipelineTemplateVersionLabel,
+      ...getGitQueryParamsWithParentScope({
+        storeMetadata: undefined,
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx:266]: this should not be undefined

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx [hunk 5]
```diff
@@ -233,18 +250,64 @@ function BarrierInputStep({ inputSetData, allowableTypes, stepViewType }: Barrie
...
+        params: projectQueryParams,
+        repoIdentifier,
+        branch,
+        loadFromFallbackBranch: true
+      })
+    },
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx:272]: please add ->

requestOptions: { headers: { 'Load-From-Cache': 'true' } }

**Comment by sahithikolichala** [comment]
retrigger jest

**Review by tan-nhu — APPROVED** [review]

**Pull request merged**